Allow AniDB as an agent for movies

Musi13 opened the pull request:
This PR adds an AniDBMovieProvider class that allows anime to be included in `Movie` type libraries. Internally, it proxies all lookups to the existing Series provider.

<details><summary>Screenshots</summary>
<p>

![anime-movies-1](https://user-images.githubusercontent.com/5846349/76157096-f72f8100-60c9-11ea-9d79-920c3ce4eed2.png)
![anime-movies-2](https://user-images.githubusercontent.com/5846349/76157097-f7c81780-60c9-11ea-961e-9e3ee4bdb05f.png)

</p>
</details>

dkanada commented:
Are movies and shows treated the same on AniDB?

Musi13 commented:
Yes, on AniDB movies are just shows with one episode. Occasionally, there's a few more "episodes" for when a movie may be split into multiple parts, but that's the general rule.

Here's an example: https://anidb.net/anime/28#Episodes

dkanada commented on Jellyfin.Plugin.Anime/Providers/AniDB/Metadata/AniDbMovieImageProvider.cs:
```diff
@@ -0,0 +1,56 @@
+namespace Jellyfin.Plugin.Anime.Providers.AniDB.Metadata
+{
+    /// <summary>
+    /// Copies the series image into a Movie, if the Movie does not otherwise have any primary image.
```
If this finds results for movies then it wouldn't be a series image.

dkanada commented on Jellyfin.Plugin.Anime/Providers/AniDB/Metadata/AniDbMovieImageProvider.cs:
```diff
@@ -0,0 +1,56 @@
+
+        public IEnumerable<ImageType> GetSupportedImages(BaseItem item)
+        {
+            return new[] { ImageType.Primary };
```
I can only see poster images, but do they have any other types?

dkanada commented on Jellyfin.Plugin.Anime/Providers/AniDB/Metadata/AniDbMovieProvider.cs:
```diff
@@ -0,0 +1,112 @@
+
+                // var episodeResult = await _episodeProvider.GetMetadata(episodeInfo, cancellationToken);
+                // if (episodeResult.HasMetadata)
+                // {
```
If you're not including it right now I would remove the unused code.

dkanada commented on Jellyfin.Plugin.Anime/Providers/AniDB/Metadata/AniDbMovieImageProvider.cs:
```diff
@@ -0,0 +1,56 @@
+            var movie = (Movie)item;
+            var seriesId = movie.ProviderIds.GetOrDefault(ProviderNames.AniDb);
+            if (string.IsNullOrEmpty(seriesId))
+                return Enumerable.Empty<RemoteImageInfo>();
```
We use brackets everywhere, so maybe add them here.

Musi13 commented:
I think I've addressed your comments. The AniDbMovieImageProvider class was just copied from the existing AniDbSeasonImageProvider class, so I didn't look at formatting.

I've reworded the summary to be a bit more clear, and changed some of the methods to more directly call the SeriesImageProvider counterparts.

dkanada commented on Jellyfin.Plugin.Anime/Providers/AniDB/Metadata/AniDbMovieImageProvider.cs:
```diff
@@ -0,0 +1,56 @@
+
+        public bool Supports(BaseItem item)
+        {
+            return item is Movie;
```
```suggestion
            return item is Movie || item is Season || item is Series;
```

dkanada replied:
I actually just checked the AniDB image providers, it looks like all of them use the exact same code. Since this interface doesn't use any generics you could probably combine them into a single `AniDbImageProvider.cs` class with the change I made above. If it doesn't work then disregard everything I said.

azagon63 replied:
It seems to work, so I've pushed the change. Anidb doesn't seem to provide any images other than the primary image, so while it looks weird that a show has the same image for the poster and all seasons, that's the only option when using AniDB as the only source.

Musi13 commented:
I also have some changes to fix/improve searching (i.e. fix #50 and add images to search results). I thought I would make another PR after this is merged since those changes are unrelated, but that would mean revving the version twice.

Is that okay, or should I add them here so everything goes in as 5.0.0?

dkanada commented:
I would revert the version bump in this one and we can merge it soon, then bump the version in your next pull request.

Musi13 commented:
Alright, I've removed the version bump from this PR and added it to a new branch. I can't PR that yet because it depends on the ImageProvider class in this one. But once this is in, I'll open the other PR.
